internal/core/command/device_test.go

```diff
@@ -15,138 +15,173 @@
 package command
 
 import (
-	"context"
 	goErrors "errors"
 	"net/http"
 	"net/url"
 	"testing"
 
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
-	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
-	"github.com/edgexfoundry/go-mod-core-contracts/models"
-	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
-
 	"github.com/edgexfoundry/edgex-go/internal/core/command/errors"
 	"github.com/edgexfoundry/edgex-go/internal/core/command/interfaces"
 	"github.com/edgexfoundry/edgex-go/internal/core/command/interfaces/mocks"
 	mdMocks "github.com/edgexfoundry/edgex-go/internal/mocks"
 	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/types"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
 )
 
 var (
-	status400 = "400"
-	status404 = "404"
-	status500 = "500"
-	status423 = "423"
+	successfulRequestStatus200   = "200"
+	badRequestStatus400          = "400"
+	notFoundStatus404            = "404"
+	internalServerErrorStatus500 = "500"
+	resourceLockedStatus423      = "423"
 
 	mismatch = "d200-c200-mismatch"
 	d200c404 = "d200-c404"
 	d200c500 = "d200-c500"
+	d200c200 = "d200c200"
 
 	TestCommandId = "TestCommandID"
 )
 
+func createTestDeviceWithPathUrl(commandID string, deviceID string) models.Device {
+
+	return models.Device{
+		AdminState: models.Unlocked,
+		Service: models.DeviceService{
+			Addressable: models.Addressable{
+				Path: "/api/v1/device/" + deviceID + "/command/" + commandID}}}
+}
+
+func createDeviceRequestWithPathParameters(
+	sampleDevice models.Device,
+	params map[string]string) *http.Request {
+
+	req, _ := http.NewRequest(http.MethodGet, cmdURI, nil)
+	req.URL.Path = sampleDevice.Service.Addressable.Path
+
+	return mux.SetURLVars(req, params)
+}
+
 // commandByDeviceID
 func TestExecuteGETCommandByDeviceIdAndCommandId(t *testing.T) {
 	tests := []struct {
 		name        string
-		deviceId    string
-		commandId   string
+		status      string
+		request     *http.Request
 		expectedErr error
 	}{
 		{
 			"Device-InvalidObjectId",
-			status400,
-			TestCommandId,
+			badRequestStatus400,
+			createDeviceRequestWithPathParameters(
+				createTestDeviceWithPathUrl(TestCommandId, badRequestStatus400),
+				map[string]string{ID: badRequestStatus400, COMMANDID: TestCommandId}),
```

Comment:
I missed this in my last comment, but can we change this `ID` variable to be something more clear. See my previous round of comments for more information.

Reply:
No worries, great feedback. I have refactored the names of this as well as the other `ID` variables for conciseness and clarity. Since refactoring variable names is necessarily subjective, should this update not align with what you have in mind, I'll be glad to refactor these variable names again so as to make the unit test as readable and as easy as possible for other users to follow along 👍